src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,565 @@
+
+        Args:
+            pixel_values_list (`List[torch.Tensor]`):
+                List of images (pixel values) to be padded. Each image should be a tensor of shape `(channels, height,
+                width)`.
+            return_tensors (`str` or [`~file_utils.TensorType`], *optional*):
+                If set, will return tensors instead of NumPy arrays. If set to `'pt'`, return PyTorch `torch.Tensor`
+                objects.
+
+        Returns:
+            [`BatchFeature`]: A [`BatchFeature`] with the following fields:
+
+            - **pixel_values** -- Pixel values to be fed to a model.
+            - **pixel_mask** -- Pixel mask to be fed to a model (when `pad_and_return_pixel_mask=True` or if
+              *"pixel_mask"* is in `self.model_input_names`).
+            - **mask_labels** -- Optional mask labels of shape `(batch_size, num_classes, height, width) to be fed to a
+              model (when `annotations` are provided).
+            - **class_labels** -- Optional class labels of shape `(batch_size, num_classes) to be fed to a model (when
+              `annotations` are provided).
+        """
+
+        max_size = self._max_by_axis([list(image.shape) for image in pixel_values_list])
+        channels, height, width = max_size
+        pixel_values = []
+        pixel_mask = []
+        mask_labels = []
+        class_labels = []
+
+        for idx, image in enumerate(pixel_values_list):
+            # create padded image
+            if pad_and_return_pixel_mask:
+                padded_image = np.zeros((channels, height, width), dtype=np.float32)
+                padded_image[: image.shape[0], : image.shape[1], : image.shape[2]] = np.copy(image)
+                image = padded_image
+            pixel_values.append(image)
+            # if we have a target, pad it
+            if annotations:
+                annotation = annotations[idx]
+                masks = annotation["masks"]
+                if pad_and_return_pixel_mask:
+                    padded_masks = np.zeros((masks.shape[0], height, width), dtype=masks.dtype)
+                    padded_masks[:, : padded_masks.shape[1], : padded_masks.shape[2]] = np.copy(padded_masks)
+                    masks = padded_masks
+                mask_labels.append(masks)
+                class_labels.append(annotation["labels"])
+            if pad_and_return_pixel_mask:
+                # create pixel mask
+                mask = np.zeros((height, width), dtype=np.int64)
+                mask[: image.shape[1], : image.shape[2]] = True
+                pixel_mask.append(mask)
+
+        # return as BatchFeature
+        data = {"pixel_values": pixel_values, "pixel_mask": pixel_mask}
+
+        if annotations:
+            data["mask_labels"] = mask_labels
+            data["class_labels"] = class_labels
+
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+
+        return encoded_inputs
+
+    def post_process_segmentation(
```

Comment:
It seems that this method is only used to postprocess the semantic segmentation outputs. Hence, I would only include 2 methods:

* post_process_semantic_segmentation
* post_process_panoptic_segmentation.

Reply:
Thanks for the comment, I prefer to leave it expose so people can have their own postprocess if needed